data-prepper-plugins/kafka-connect-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafkaconnect/source/MongoDBSource.java

```diff
@@ -5,28 +5,123 @@
 import org.opensearch.dataprepper.model.configuration.PipelineDescription;
+import org.opensearch.dataprepper.model.record.Record;
 import org.opensearch.dataprepper.model.source.Source;
+import org.opensearch.dataprepper.model.source.coordinator.SourceCoordinator;
+import org.opensearch.dataprepper.model.source.coordinator.UsesSourceCoordination;
 import org.opensearch.dataprepper.plugins.kafka.extension.KafkaClusterConfigSupplier;
 import org.opensearch.dataprepper.plugins.kafkaconnect.configuration.MongoDBConfig;
 import org.opensearch.dataprepper.plugins.kafkaconnect.extension.KafkaConnectConfigSupplier;
+import org.opensearch.dataprepper.plugins.kafkaconnect.source.mongoDB.MongoDBService;
+import org.opensearch.dataprepper.plugins.kafkaconnect.source.mongoDB.MongoDBSnapshotProgressState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
 
 /**
  * The starting point of the mysql source which ingest CDC data using Kafka Connect and Debezium Connector.
  */
 @SuppressWarnings("deprecation")
 @DataPrepperPlugin(name = "mongodb", pluginType = Source.class, pluginConfigurationType = MongoDBConfig.class)
-public class MongoDBSource extends KafkaConnectSource {
+public class MongoDBSource extends KafkaConnectSource implements UsesSourceCoordination {
+    private static final Logger LOG = LoggerFactory.getLogger(MongoDBSource.class);
+    private static final String COLLECTION_SPLITTER = "\\.";
+
+    private final AwsCredentialsSupplier awsCredentialsSupplier;
+
+    private final PluginMetrics pluginMetrics;
+
+    private final AcknowledgementSetManager acknowledgementSetManager;
+
+    private MongoDBService mongoDBService;
+
+    private SourceCoordinator<MongoDBSnapshotProgressState> sourceCoordinator;
+
+    private ByteDecoder byteDecoder;
 
     @DataPrepperPluginConstructor
     public MongoDBSource(final MongoDBConfig mongoDBConfig,
                          final PluginMetrics pluginMetrics,
                          final PipelineDescription pipelineDescription,
+                         final AcknowledgementSetManager acknowledgementSetManager,
+                         final AwsCredentialsSupplier awsCredentialsSupplier,
                          final KafkaClusterConfigSupplier kafkaClusterConfigSupplier,
                          final KafkaConnectConfigSupplier kafkaConnectConfigSupplier) {
         super(mongoDBConfig, pluginMetrics, pipelineDescription, kafkaClusterConfigSupplier, kafkaConnectConfigSupplier);
+        this.pluginMetrics = pluginMetrics;
+        this.acknowledgementSetManager = acknowledgementSetManager;
+        this.awsCredentialsSupplier = awsCredentialsSupplier;
+        this.byteDecoder = new JsonDecoder();
+        this.validateCollections();
+    }
+
+    @Override
+    public void start(Buffer<Record<Object>> buffer) {
+        super.start(buffer);
+        if (shouldStartInitialLoad()) {
+            LOG.info("Starting initial load");
```

Comment:
initial load of what? Just the export? Could maybe add some description here

Reply:
Initial load of existing mongoDB data. It is different with DynamoDB's export. Calling export just to make it similar for users.